eng/pipelines/templates/jobs/build.yml

```diff
@@ -76,6 +76,23 @@ jobs:
     inputs:
       summaryFileLocation: $(CoverageFile)
 
+  - bash: |
+      if [ "$(uname -m)" = "x86_64" ]; then
+        # x64 host building AOT targeting arm64 need cross-compilation toolchain.
+        sudo apt-get update
+        sudo apt-get install -y gcc-aarch64-linux-gnu g++-aarch64-linux-gnu libc6-dev-arm64-cross binutils-aarch64-linux-gnu
+
+        # make the generic 'objcopy' point to the 'objcopy' from the arm64 cross-compilation toolchain.
+        sudo ln -sf /usr/bin/aarch64-linux-gnu-objcopy /usr/local/bin/objcopy
+
+        export CC=aarch64-linux-gnu-gcc
```

Comment:
I don't see any docs on the Bash task that would suggest these environment variables are persisted after the task completes.   If they should be available to other tasks in the job, you'll need to set devops variables via echo/Write-Host statements:

```
echo "##vso[task.setvariable variable=CC;]aarch64-linux-gnu-gcc"
```

When imported to following tasks, the environment variable name will always be upper case.

Reply:
changed the scripts entirely to follow the official doc from .NET AOT team https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/cross-compile#linux